### ISSUE TITLE
in rare cases, NSCBC outflow produces unphysically high velocities

### ISSUE THREAD

**Issue by BenWibking** [quokka-astro/quokka#532]
**Describe the bug**
In very rare circumstances, the NSCBC outflow boundary conditions can produce states in isolated cells at the boundary that have unphysically high velocities. An ad hoc (but partially effective) workaround is to revert to zero-gradient extrapolation when the velocity in the ghost cells would otherwise be extremely large (for example, > 1e10 cm/s).

**To Reproduce**
This was observed after about 40,000 timesteps while running the updated shock-cloud problem at 1024x256x256 resolution with no refinement.

**Additional context**
A similar failure mode (but with negative densities instead) was seen in https://github.com/quokka-astro/quokka/pull/472.

**Comment by psharda** [comment]
@BenWibking is this now resolved by #598 ?

**Comment by BenWibking** [comment]
> @BenWibking is this now resolved by #598 ?

Yes, in the sense that the simple version of the boundary conditions added in #598 doesn't have this problem. No, in the sense that the real NSCBC still has this problem.

**Comment by BenWibking** [comment]
Applying a slope limiter to the primitive var data used as input to the NSCBC algorithm may help here.
